### PR TITLE
[KOGITO-2511] Changes in Schema generation

### PR DESCRIPTION
**Pull request by fjtirado**
Adding "kogito.jsonSchema.version" property for maven and quarkus
Adding "date-time" format for Date types.

Many thanks for submitting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [ ] You have read the [contributors guide](CONTRIBUTING.md)
- [ ] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [ ] Pull Request title contains the target branch if not targeting master: `[0.9.x] KOGITO-XYZ Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains link to any dependent or related Pull Request
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket

### CODE REVIEW

**Comment by evacchi** [comment]
@radtriste @MarianMacik can either review ack this?

File: kogito-codegen/src/main/java/org/kie/kogito/codegen/JsonSchemaGenerator.java [hunk 1]
```diff
@@ -65,10 +71,22 @@ public Builder withGenSchemaPredicate(Predicate<? super Class<?>> shouldGenSchem
...
+
         public JsonSchemaGenerator build() {
             JsonSchemaGenerator instance = new JsonSchemaGenerator(stream);
             instance.getSchemaName = getSchemaName != null ? getSchemaName : JsonSchemaGenerator::getKey;
             instance.shouldGenSchema = shouldGenSchema != null ? shouldGenSchema : JsonSchemaGenerator::isUserTaskClass;
+            if (schemaVersion != null) {
+                try {
+                    instance.schemaVersion = SchemaVersion.valueOf(schemaVersion.trim().toUpperCase());
+                } catch (IllegalArgumentException ex) {
+                    logger.warn("Invalid value {} for schema version. Using default", schemaVersion);
```
Comment (MarianMacik) [inline kogito-codegen/src/main/java/org/kie/kogito/codegen/JsonSchemaGenerator.java:87]: What about throwing even an exception here? I think one warning can be lost in the logs and users can then expect a different schema version to be used than it actually is. Though I agree this is a matter of preference.

File: kogito-codegen/src/test/java/org/kie/kogito/codegen/JsonSchemaGeneratorTest.java [hunk 2]
```diff
@@ -84,7 +88,25 @@ public void testJsonSchemaGenerator() throws IOException {
         assertEquals(1, files.size());
         GeneratedFile file = files.iterator().next();
         assertEquals("test_test.json", file.relativePath());
-        assertSchema(file);
+        assertSchema(file, SchemaVersion.DRAFT_2019_09);
+    }
+    
+    @Test
+    public void testJsonSchemaGeneratorNonDraft() throws IOException {
```
Comment (MarianMacik) [inline kogito-codegen/src/test/java/org/kie/kogito/codegen/JsonSchemaGeneratorTest.java:95]: A little nitpick. The name of the tests could me more descriptive -> `testJsonSchemaGeneratorNonExistingDraft`.

Comment (fjtirado) [reply kogito-codegen/src/test/java/org/kie/kogito/codegen/JsonSchemaGeneratorTest.java]: Good catch, I missed a word

File: kogito-codegen/src/main/java/org/kie/kogito/codegen/JsonSchemaGenerator.java (hunk 1 above)
Comment (fjtirado) [reply kogito-codegen/src/main/java/org/kie/kogito/codegen/JsonSchemaGenerator.java]: Your approach is probably better, so changed ;)

**Pull request merged**